webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
-            }
-          }
+    ArrayList<String> arrayToClassify = new ArrayList<>();
+    if (publish) {
+      arrayToClassify.add(id);
+      scc.publish(arrayToClassify); // set etat du site a 1
+    }
+  }
 
-          SiteDetail ancien = scc.getWebSite(id);
-          int type = ancien.getType();
+  private String classification(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
 
-          /* update description en BD */
-          SiteDetail descriptionSite2 =
-              new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, type, null,
-                  null, etat, popup);
+    String action = request.getParameter("Action");
+    String id = request.getParameter("TopicId");
+    String linkedPathString = request.getParameter("Path");
 
-          scc.updateWebSite(descriptionSite2);
+    Collection<SiteDetail> listeSites = scc.getAllWebSite();
+    request.setAttribute("ListSites", listeSites);
+    request.setAttribute("CURRENT_FOLDER_PARAM", scc.getSessionTopic());
 
-          if (listIcons.size() > 0) {
-            scc.associateIcons(id, listIcons);
-          }
+    return
+        "/webSites/jsp/classifyDeclassify.jsp?Action=" + action + "&TopicId=" + id + "&Path=" +
+            linkedPathString;
+  }
 
-          /* publications : classer le site dans les themes cochés */
-          ArrayList<String> arrayToClassify = new ArrayList<>();
-          boolean publish = false;
-          ArrayList<String> arrayTopic = new ArrayList<>();
-          begin = 0;
-          end = 0;
-          end = listeTopics.indexOf(',', begin);
-          String idTopic;
-          while (end != -1) {
-            idTopic = listeTopics.substring(begin, end);
-
-            begin = end + 1;
-            end = listeTopics.indexOf(',', begin);
-
-            arrayTopic.add(idTopic);
-            publish = true;
-          }
+  private String updateTopic(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
 
-          scc.updateClassification(id, arrayTopic);
+    String id = request.getParameter("ChildId");
+    String path = request.getParameter("Path");
 
-          arrayToClassify.add(id);
-          if (publish) {
-            scc.publish(arrayToClassify); // set etat du site a 1
-          } else {
-            scc.dePublish(arrayToClassify);
-          }
-        }
+    NodeDetail folderDetail = scc.getFolderDetail(id);
+    request.setAttribute("CURRENT_FOLDER_PARAM", folderDetail);
+
+    return "/webSites/jsp/updateTopic.jsp?ChildId=" + id + "&Path=" + path;
+  }
 
-        Collection<SiteDetail> listeSites = scc.getAllWebSite();
-        request.setAttribute("ListSites", listeSites);
-        request.setAttribute("BookmarkMode", Boolean.valueOf(scc.isBookmarkMode()));
+  private String addTopic(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "userRole". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)